### DISCUSSION TITLE
Xamarin Forms to MAUI migration

### DISCUSSION THREAD

**Discussion by IXChinmayD** [PrismLibrary/Prism#3087] [.NET MAUI]
I wanted to migrate from Xamarin Forms to the MAUI .NET 8 mobile application. We have used Prism for View ViewModel binding and 
Page Navigation using Prism Packages.

1. I cannot find proper documentation for the migration to .NET MAUI 8.  
2. Which packages do I need to use for the migration? What is the difference between Prism?MAUI vs Prism.MAUI.DryIoc? I can see the pre-release for the same which is the latest stable package for the same?
![Screenshot 2024-02-26 120749](https://github.com/PrismLibrary/Prism/assets/155611333/5ca31cc1-e2fd-4df2-a355-d745c824150c)
3.  I am getting the same error as mentioned in the stack overflow link:- https://stackoverflow.com/questions/76273297/maui-android-either-set-mainpage-or-override-createwindows

**Comment by dansiegel** [comment]
1) Docs are located at https://docs.prismlibrary.com. 
2) The difference is the same as it was for Xamarin.Forms. Prism.Maui is the core library for the Platform while the DryIoc package is the one that has an implemented DI container.

| Xamarin.Forms | .NET MAUI |
|---------------|-----------|
| Prism.Forms | Prism.Maui |
| Prism.DryIoc.Forms | Prism.DryIoc.Maui |
| Prism.Unity.Forms | n/a -> use Prism.DryIoc.Maui |

3) There is so much wrong with the code in that StackOverflow I'm not even sure where to begin. If you used the earlier previews you will need to migrate OnAppStart to CreateWindow. You should neither be setting MainPage or overriding the CreateWindow in the Application. There is also no reason to resolve the NavigationService with the container when the method already provides you the NavigationService. Please refer to the e2e app for MAUI for an example of how to set this up.

**Reply by IXChinmayD** [reply]
Thank you for the reply. According to my company policy compliance, we cannot use the pre-release versions. When can we expect the production-ready release of the Prism?
Also in Xamarin Forms, the static variables of App.xaml.cs were initialized in the OnInitialise() override methods of the same class along with some methods that initialize some processing running in the background while the user is using the application. I have attached the code snippet image for you to look over. Where should I write the below code in MAUI?

![image](https://github.com/PrismLibrary/Prism/assets/155611333/722cc4e0-61c7-43b9-aae1-b9711cfa7658)

**Reply by dansiegel** [reply]
Please use the E2E testing app as a guide:
https://github.com/PrismLibrary/Prism/tree/master/e2e/Maui/PrismMauiDemo

**Reply by IXChinmayD** [reply]
Thank you for your quick response.

**Comment by giuseppenovielli** [comment]
These are samples, but are as a docs. Check it.

https://github.com/giuseppenovielli/Prism-MAUI-Samples

I will update to last Prism version and .NET MAUI8, but the basic structure is there.
